packages/styled-components/src/models/StyledNativeComponent.js

```diff
@@ -69,6 +70,12 @@ class StyledNativeComponent extends Component<*, *> {
             return [generatedStyles].concat(style(state))
           }
           : [generatedStyles].concat(style);
+          if (isArrayEqual(propsForElement.style, this.previousStyle)) {
```

Comment:
Would recommend if we could add a small comment here specifying why we are doing it.

Reply:
Added